src/components/SearchBar.tsx

```diff
@@ -1,71 +1,68 @@
-import React, { useState, useEffect } from 'react';
+import React, { useEffect } from 'react';
 
-import { Button } from './UI/Button';
 import { Input } from './UI/Input';
-import { NetworkSelector } from './NetworkSelector';
+import { Dropdown } from './UI/Dropdown';
 import { logNetworkChange } from '../shared/logs';
-import { usePersistentState } from '../shared/usePersistentState';
 import { getActiveTabUrl } from '../helpers/getActiveTab';
-import { Radio } from './UI/Radio';
+import { RadioGroup, RadioOption } from './UI/Radio';
 
-import { Severity } from '../types/api';
+import { networkDropdownOptions } from '../types/networks';
 
 import * as Styled from './SearchBar.styles';
 
-interface SearchBarProps {
-    onAddressClick: (chainId: string, to: string) => void;
-    onURLClick: (url: string) => void;
-    disabled: boolean;
-    severity: undefined | Severity;
-    compact?: boolean;
-    persistent?: boolean;
+export enum SearchMode {
+    Address = 'address',
+    URL = 'url',
 }
 
-export function SearchBar({
-    onAddressClick,
-    onURLClick,
-    disabled,
-    severity,
-    compact,
-    persistent = true,
-}: SearchBarProps) {
-    const [address, setAddress] = useState('');
-    const [chainId, setChainId] = persistent ? usePersistentState<string>('chainId', '0x1') : useState('0x1');
-    const [url, setUrl] = useState('');
-    const [isAddress, setIsAddress] = useState(true);
+export type SearchState = {
+    mode: SearchMode;
+    chainId: string;
```

Comment:
why not use the enum for the chainID?
`chainId: keyof typeof SupportedNetworks; `